sdk/src/transaction_context.rs

```diff
@@ -32,7 +32,8 @@ pub struct TransactionContext {
     instruction_context_capacity: usize,
     instruction_context_stack: Vec<InstructionContext>,
     number_of_instructions_at_transaction_level: usize,
-    instruction_trace: Vec<Vec<CompiledInstruction>>,
+    top_level_instruction_trace: Vec<InstructionContext>,
```

Comment:
This is close to what I had in mind. Why not include `top_level_instruction_trace` as the first entry in each `inner_instruction_trace`? `deconstruct()` in bank.rs would then need to skip the first entry, but it might be a bit more compact and easier to understand.

(I can also refactor that later myself)

Reply:
Updated, check it out